### DISCUSSION TITLE
Contradictory samples? Decremental learning?

### DISCUSSION THREAD

**Discussion by FedericoMz** [online-ml/river#1216] [Q&A]
Let's say in my data stream, there's a sample (x, y1). Later, there's (x, y2). Same sample as before, but with a different label. We can assume that the later sample has the "correct" label, for simplicity.

How can I deal with this case with River? Looking at the documentation, I don't think it is possible to "detrain" a model, as decremental learning is not supported. Is there any alternative? Perhaps training the model _twice_ on (x, y2) to reduce (x, y1) importance?

**Comment by MaxHalford** [comment]
Hey there 👋

There is no explicit way to "unlearn" a sample. It would be too complicated.

The problem you describe is connected to the notion of concept drift. As the data stream moves on, the distribution of the data changes. That's one of the advantages online learning offers: it adapts.

The more data you show the model, the more it will forget about past data. This happens naturally with any online learning algorithm. The pace at which the model forgets depends on the model's and it's parameters. For instance, increasing the learning rate in a linear model increases the pace.

Hope this helps.

**Reply by FedericoMz** [reply]
Thank you for the answer (and the fantastic library; I just started with online learning, and everything is flawless)!

Do you think I should use an algorithm with built-in drift detection for my use case? I _know_ that some contradictory samples are likely to show up in my data stream, but I don't know in advance when.

**Reply by MaxHalford** [reply]
Honestly I'm not sure. But probably the answer is yes. 

I'm sure you can quickly answer that by experimenting by yourself. Let us know what works best!
